export/client/clients/database-client.go

```diff
@@ -0,0 +1,131 @@
+/*******************************************************************************
+ * Copyright 2018 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * @microservice: export-client-go library
+ * @author: Ryan Comer, Dell
+ * @version: 0.5.0
+ *******************************************************************************/
+package clients
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/edgexfoundry/edgex-go/export"
+	"gopkg.in/mgo.v2/bson"
+)
+
+type DatabaseType int8 // Database type enum
+const (
+	INVALID DatabaseType = iota
+	MONGO
+	MOCK
+)
+
+const (
+	invalidStr = "invalid"
+	mongoStr   = "mongodb"
+	mockStr    = "mockdb"
+)
+
+// Add in order declared in Struct for string value
+var databaseArr = [...]string{invalidStr, mongoStr, mockStr}
+
+func (db DatabaseType) String() string {
+	if db >= INVALID && db <= MOCK {
+		return databaseArr[db]
+	}
+	return invalidStr
+}
+
+// Return enum value of the Database Type
+func GetDatabaseType(db string) DatabaseType {
+	if mongoStr == db {
```

Comment:
`switch` would be a better construct here

Reply:
Done!